tests/rgbpp/xudt/xudt-on-ckb/2-transfer-xudt.ts

```diff
@@ -0,0 +1,157 @@
+
+    const outputs: CKBComponents.CellOutput[] = receivers.map(({ toAddress }) => ({
+      lock: addressToScript(toAddress),
+      type: xudtType,
+      capacity: append0x(calculateUdtCellCapacity(addressToScript(toAddress)).toString(16)),
+    }));
+    const outputsData = receivers.map(({ transferAmount }) => append0x(u128ToLe(transferAmount)));
+
+    if (sumAmount > sumTransferAmount) {
+      const xudtChangeCapacity = calculateUdtCellCapacity(fromLock);
+      outputs.push({
+        lock: fromLock,
+        type: xudtType,
+        capacity: append0x(xudtChangeCapacity.toString(16)),
+      });
+      outputsData.push(append0x(u128ToLe(sumAmount - sumTransferAmount)));
+      sumXudtOutputCapacity += xudtChangeCapacity;
+    }
+
+    const txFee = MAX_FEE;
+    if (sumXudtInputsCapacity <= sumXudtOutputCapacity) {
+      let emptyCells = await collector.getCells({
+        lock: fromLock,
+      });
+      if (!emptyCells || emptyCells.length === 0) {
+        throw new NoLiveCellError('The address has no empty cells');
+      }
+      emptyCells = emptyCells.filter((cell) => !cell.output.type);
+      const needCapacity = sumXudtOutputCapacity - sumXudtInputsCapacity;
+      const { inputs: emptyInputs, sumInputsCapacity: sumEmptyCapacity } = collector.collectInputs(
+        emptyCells,
+        needCapacity,
+        txFee,
+        { minCapacity: MIN_CAPACITY },
+      );
+      inputs = [...inputs, ...emptyInputs];
+      actualInputsCapacity += sumEmptyCapacity;
+    }
+
+    let changeCapacity = actualInputsCapacity - sumXudtOutputCapacity;
+    outputs.push({
+      lock: fromLock,
+      capacity: append0x(changeCapacity.toString(16)),
+    });
+    outputsData.push('0x');
+
+    const emptyWitness = { lock: '', inputType: '', outputType: '' };
+    const witnesses = inputs.map((_, index) => (index === 0 ? emptyWitness : '0x'));
+
+    const cellDeps = [getSecp256k1CellDep(isMainnet), ...(await fetchTypeIdCellDeps(isMainnet, { xudt: true }))];
+
+    const unsignedTx = {
+      version: '0x0',
+      cellDeps,
+      headerDeps: [],
+      inputs,
+      outputs,
+      outputsData,
+      witnesses,
+    };
+
+    if (txFee === MAX_FEE) {
+      const txSize = getTransactionSize(unsignedTx) + SECP256K1_WITNESS_LOCK_SIZE;
+      const estimatedTxFee = calculateTransactionFee(txSize);
+      changeCapacity -= estimatedTxFee;
+      unsignedTx.outputs[unsignedTx.outputs.length - 1].capacity = append0x(changeCapacity.toString(16));
+    }
+
+    const signedTx = collector.getCkb().signTransaction(CKB_PRIVATE_KEY)(unsignedTx);
+    const txHash = await collector.getCkb().rpc.sendTransaction(signedTx, 'passthrough');
+
+    console.info(`xUDT asset has been minted or transferred and tx hash is ${txHash}`);
+    console.info(`explorer: https://pudge.explorer.nervos.org/transaction/${txHash}`);
+  });
+};
+
+const XUDT_TOKEN_INFO: RgbppTokenInfo = {
+  decimal: 8,
+  name: 'XUDT Test Token',
+  symbol: 'PDD',
```

Comment:
The xUDT issued by the `fromLock` is [symbol: ccli](https://pudge.explorer.nervos.org/xudt/0x14f4972e17dd926f7ce55b3df5b1a31ba6e669a4ea7846e4f064535c1f50dac3) not `PDD`.


> [!NOTE]
> Only the first uniue cell combined with the xUDT in the same transaction is valid.
> a.k.a. The first transaction containing a unique cell will definitely declare the corresponding xUDT.
> See https://pudge.explorer.nervos.org/xudt/0x14f4972e17dd926f7ce55b3df5b1a31ba6e669a4ea7846e4f064535c1f50dac3

Reply:
### What I defined in the integration test is 'pdd'
https://github.com/ckb-cell/rgbpp-sdk/blob/2d634abae666610d53de03a9664d3e43fee1bef3/tests/rgbpp/xudt/xudt-on-ckb/1-issue-xudt.ts#L115C1-L117C3

Reply:
The same owner lock script is used to issue multiple xUDTs whose type scripts args are owner lock hash.

So the second issue transaction (PDD) will be regarded as re-issuing CCLI for CKB explorer.

Reply:
> ### What I defined in the integration test is 'PDD'

See this tx: [0x9af4e6afebb377f733fff632094722b246825888da49af7cf5c30d926a27a4c9](https://pudge.explorer.nervos.org/transaction/0x9af4e6afebb377f733fff632094722b246825888da49af7cf5c30d926a27a4c9)

The xUDT was first defined as `ccli` on 2024/05/30 17:11:12

Reply:
> See this tx: [0x9af4e6afebb377f733fff632094722b246825888da49af7cf5c30d926a27a4c9](https://pudge.explorer.nervos.org/transaction/0x9af4e6afebb377f733fff632094722b246825888da49af7cf5c30d926a27a4c9)
> 
> The xUDT was first defined as `ccli` on 2024/05/30 17:11:12

Please check this transaction. 

https://pudge.explorer.nervos.org/transaction/0x6a35da16ab1198008545c78b91abe22999f0dc823055553a13d7de29f3063111
The example you provided is not the same lock.

Reply:
Ok, you are right.